### ISSUE TITLE
keycloak_openid_client_scope is not setting Assigned Type field

### ISSUE THREAD

**Issue by tp20165** [keycloak/terraform-provider-keycloak#776]
In the Master realm, I am generating client scope using keycloak_openid_client_scope resource. It created scope with Assigned Type as None. I need to set it as Default. I don't see any option to set the assigned type. Am I missing anything?

**Comment by mrohlof-protofy** [comment]
I need this, too. Seems like in https://github.com/mrparkers/terraform-provider-keycloak/blob/10e59625b45741464ca3c34a3fba07011981e6a5/provider/resource_keycloak_openid_client_scope.go, the `type` field is missing. 

This way, we can not create scopes that are default for every new client - which is a pain.

**Comment by lesaux** [comment]
keycloak_openid_client_default_scopes seems to have no effect for me.

The Assigned type for a custom keycloak_openid_client_scope remains to "None" in the GUI, despite the terraform apply running fine.

**Comment by francoisauclair911** [comment]
Just checking if there was some progress on this ?

Thanks!

**Comment by hameno** [comment]
We also need this...

**Comment by sschu** [comment]
As noted here (https://github.com/keycloak/terraform-provider-keycloak/issues/881#issuecomment-2665078884), this works using separate resources.

**Comment by datenzar** [comment]
Please correct me if I'm wrong here but from what I see, is that scopes listed in `keycloak_openid_client_default_scopes` and `keycloak_openid_client_optional_scopes` are only applied to clients which are created thereafter but not to existing clients. In order to take effect, you would need to recreate all clients or apply the scopes manually (which is what we're trying to avoid by using TF in the first place ;-) ).

**Comment by sschu** [comment]
@datenzar But isnt this basically the behaviour of Keycloak or is this different if you do it via the provider vs. doing it in Keycloak directly?

**Comment by datenzar** [comment]
Yes, it reflects absolutely the behavior in the sense that the Assigned Type of a Realm Client Scope is applied when creating a new OpenID client.

Assume the following scenario: I have a client scope which I want to define inside the realm a set of OpenID clients which are already in place. Currently I have no way in Terraform to achieve it. 

What I can do instead is define this client scope as default and delete and recreate the clients. What I currently cannot do, is setting the assigned scope on the client directly (e.g. as an argument to the keycloak_openid_client Resource).

**Comment by datenzar** [comment]
You're right, I must have overlooked this resource. Thanks for pointing me to the right place!
